knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -361,21 +361,21 @@ def updatePart(df, ns, ne):
                 print("error from:{} to :{}".format(ns, ne))
 
         # Cross join Data
-        O_gdf = left_input.to_pandas().rename(columns={self.O_geo_col: "geometry"})
-        D_gdf = right_input.to_pandas().rename(columns={self.D_geo_col: "geometry"})
-        O_gdf = gp.GeoDataFrame(O_gdf, geometry="geometry")
-        D_gdf = gp.GeoDataFrame(D_gdf, geometry="geometry")
+        O_gdf = knut.load_geo_data_frame(left_input, self.O_geo_col, exec_context)
+        D_gdf = knut.load_geo_data_frame(right_input, self.D_geo_col, exec_context)
+        O_gdf = O_gdf[[self.O_geo_col]].rename(columns={self.O_geo_col: "geometry"})
```

Comment:
as the naming will influence the subsequent names generated by spatial join result.  If sometimes both are 'geometry' but sometimes they are different, it might need additional handling. So I just keep it